var/spack/repos/builtin/packages/qthreads/package.py

```diff
@@ -29,13 +29,23 @@ class Qthreads(AutotoolsPackage):
 
     homepage = "http://www.cs.sandia.gov/qthreads/"
 
-    url = "https://github.com/Qthreads/qthreads/releases/download/1.10/qthread-1.10.tar.bz2"
+    url = "https://github.com/Qthreads/qthreads/releases/download/1.18/qthread-1.18.tar.gz"
     test_requires_compiler = True
     test_base_path = "test/basics/"
     test_list = ["hello_world_multi", "hello_world"]
 
     tags = ["e4s"]
 
+    def url_for_version(self, version):
+        # if version is greater than 1.17, use new default
+        if version >= Version("1.17"):
+            return super(Qthreads, self).url_for_version(version)
+        # otherwise, use .bz2 format
+        url_fmt = "https://github.com/Qthreads/qthreads/releases/download/{0}/qthread-{0}.tar.bz2"
+        return url_fmt.format(version)
+
```

Comment:
Please move this so it is the first method (i.e., after the directives).

Reply:
I moved the function to the top of the class after the homepage directive. Is that what you meant here?

Reply:
Sorry, I misunderstood. I have moved it again. Is it correct currently?

Reply:
No worries.